common/components/common/projects/EditProjectForm.jsx

```diff
@@ -102,50 +115,87 @@ class EditProjectForm extends React.PureComponent<Props,State> {
           project_organization: project.project_organization,
           project_issue_area: project.project_issue_area,
           project_technologies: project.project_technologies,
-          project_links: _.cloneDeep(project.project_links),
           project_files: _.cloneDeep(project.project_files),
           project_positions: _.cloneDeep(project.project_positions),
-          project_thumbnail: project.project_thumbnail
+          project_thumbnail: project.project_thumbnail,
         }
       });
+      //this will set formFields.project_links and formFields.links_*
+      this.filterSpecificLinks(_.cloneDeep(project.project_links));
     }
   }
-  
+
   handleLoadProjectError(error: APIError): void {
     this.setState({
       error: "Failed to load project information"
     });
   }
-  
+
   onFormFieldChange(formFieldName: string, event: SyntheticInputEvent<HTMLInputElement>): void {
     this.state.formFields[formFieldName] = event.target.value;
     this.forceUpdate();
   }
-  
+
   onTagChange(formFieldName: string, value: $ReadOnlyArray<TagDefinition>): void {
     this.state.formFields[formFieldName] = value;
   }
-  
+
   onSubmit(): void {
     //Sanitize project url if necessary
     if(this.state.formFields.project_url) {
       this.state.formFields.project_url = url.appendHttpIfMissingProtocol(this.state.formFields.project_url);
     }
-    this.forceUpdate();
+   // create input array
+   var eLinks = ['link_coderepo','link_messaging','link_filerepo','link_projmanage'].map(name => ({linkName: name, linkUrl: this.state.formFields[name]}))
+   //create output array
+   var eLinksArray = [];
+  //create objects for project_links array, skipping empty fields
+    eLinks.forEach(function(item) {
+      if(!_.isEmpty(item.linkUrl)) {
+        eLinksArray.push({
+          linkName: item.linkName,
+          linkUrl: item.linkUrl,
+          visibility: "PUBLIC",
+        })
+      }
+    });
+    //combine arrays prior to sending to backend
+    var combinedArray = this.state.formFields.project_links.concat(eLinksArray);
+    // setState new combined array
+    this.setState({ formFields: { project_links: combinedArray }});
   }
-  
+
+  filterSpecificLinks(array) {
+    //this function updates the entire state.formFields object at once
+    var specificLinks = _.remove(array, function(n) {
+      return n.linkName.startsWith("link_")
```

Comment:
To make it more clear, you can replace this with
return n.linkName in linkNames;